tests/bwc/test_rolling_upgrade.py

```diff
@@ -101,7 +83,12 @@ def _test_rolling_upgrade(self, path, nodes):
             cluster[idx] = new_node
             with connect(new_node.http_url, error_trace=True) as conn:
                 c = conn.cursor()
-                wait_for_active_shards(c, expected_active_shards)
+
+                c.execute("select decisions['explanations'] from sys.allocations") # wait_for_active_shards assertion fails, let's get more info
+                res = c.fetchall()
+                self.assertEqual(res, [["dummy"]]) # dummy equal, see assertion error to get real info
+
```

Comment:
```
First differing element 0:
[[None, None]]
['dummy']

First list contains 8 additional elements.
First extra element 1:
[[None, None, None]]

+ [['dummy']]
- [[[None, None]],
-  [[None, None, None]],
-  [[None, None]],
-  [[None,
-    ['a copy of this shard is already allocated to this node [[t1][2], '
-     'node[scE2yZHeSlKCbYNT0gOQZQ], [P], s[STARTED], '
-     'a[id=cWMNjpfNQTuBQIxJqS0Xmw]]']]],
-  [[None,
-    ['a copy of this shard is already allocated to this node [[t1][2], '
-     'node[1k8WhMc1Q_KUkH0xhzCJgg], [R], s[STARTED], '
-     'a[id=1IDCucQ1TQqgdXDiDu3LMQ]]']]],
-  [[None, None]],
-  [[None,
-    None,
-    ['a copy of this shard is already allocated to this node [[t1][1], '
-     'node[scE2yZHeSlKCbYNT0gOQZQ], [P], s[STARTED], '
-     'a[id=vJvIa_TAST2Pq7ORK_TzUQ]]']]],
-  [[None, None]],
-  [[None,
-    None,
-    ['a copy of this shard is already allocated to this node [[t1][0], '
-     'node[1k8WhMc1Q_KUkH0xhzCJgg], [P], s[STARTED], '
-     'a[id=sWk5xoYLRcWfAKNRwXQl9Q]]']]]]

```

but it's not same with reported 
`shard has exceeded the maximum number of retries [5] on failed allocation attempts - manually execute 'ALTER CLUSTER REROUTE RETRY FAILED' to retry, [unassigned_info[[reason=ALLOCATION_FAILED], at[2024-07-17T11:47:31.245Z], failed_attempts[5], failed_nodes[[msx5erdrReW2S6nK1ytP9g]], delayed=false, details[failed shard on node [msx5erdrReW2S6nK1ytP9g]: failed to create index, failure java.lang.IllegalArgumentException: unknown setting [index.warmer.enabled]`

Reply:
but `wait_for_active_shards` failure is probably already good indicator of some issue